packages/xo-server-perf-alert/src/index.js

```diff
@@ -524,11 +524,39 @@ ${monitorBodies.join('\n')}`
                   })
                 }
 
-                result.listItem = `  * ${result.objectLink}: ${
-                  result.value === undefined
-                    ? "**Can't read performance counters**"
-                    : result.value.toFixed(1) + typeFunction.unit
-                }\n`
+                const getListItem = () => {
+                  if (result.value === undefined) {
+                    return "**Can't read performance counters**"
+                  }
+
+                  if (lcObjectType === 'vm' && definition.variableName === 'memoryUsage') {
+                    const checkManagementAgent = (vm, guestMetrics) => {
+                      if ((vm.power_state !== 'Running' && vm.power_state !== 'Paused') || guestMetrics === undefined) {
+                        return
+                      }
+
+                      const { major, minor } = guestMetrics.PV_drivers_version
+                      const hasPvVersion = major !== undefined && minor !== undefined
+                      return hasPvVersion || guestMetrics.other['feature-static-ip-setting'] === '1'
+                    }
+
+                    const vm = result.object
+                    const guestMetrics = this._xo.getXapi(uuid).getObject(vm.guest_metrics)
+
+                    const managementAgentDetected = checkManagementAgent(vm, guestMetrics)
+
+                    if (managementAgentDetected === undefined) {
+                      return "**Can't read performance counters**"
+                    }
+                    if (managementAgentDetected === false) {
+                      return '**Guest tools must be installed**'
+                    }
+                  }
+
+                  return result.value?.toFixed(1) + typeFunction.unit
```

Comment:
`result.value` can't be `undefined`. Can it be `null` or something else? And if so, does it make sense to show `"undefined"` in the report?

Reply:
It can be undefined before the object is retrieved and it provokes an error "ReferenceError: result is not defined"

Reply:
At the beginning of `getListItem`, you check if `result.value === undefined` and return `**Can't read performance counters**.`

So, if `result.value` is `undefined`, you shouldn't be reaching this code.

IMHO, if `result.value` can be `null`, you should update your `if (result.value === undefined)` to `if (result.value == undefined)`.